rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/InlineItemModificationCreator.java

```diff
@@ -0,0 +1,171 @@
+
+    public static final String EDIT_FORM_ITEM_SUFFIX = ".edit";
+
+    public static final String EDIT_RADIO_OPTION_NAME = "edit";
+
+    public static final String UPLOAD_RADIO_OPTION_NAME = "upload";
+
+    private final NodeSourceWindow nodeSourceWindow;
+
+    public InlineItemModificationCreator(NodeSourceWindow nodeSourceWindow) {
+        this.nodeSourceWindow = nodeSourceWindow;
+    }
+
+    public List<FormItem> getModificationChoiceItemsForNonTextualFields(PluginDescriptor plugin,
+            PluginDescriptor.Field pluginField) {
+        FormItem chooseCredentialsFormItem;
+        List<FormItem> formItemsReplacingNonTextualFormItems = new LinkedList<FormItem>();
+
+        if (plugin.getPluginName().equals(nodeSourceWindow.getFocusedInfrastructurePluginName()) ||
+            plugin.getPluginName().equals(nodeSourceWindow.getFocusedPolicyPluginName())) {
+
+            RadioGroupItem editOrUploadFormItem = createRadioItemToModifyPluginField(plugin, pluginField);
+            formItemsReplacingNonTextualFormItems.add(editOrUploadFormItem);
+
+            TextAreaItem previousValueItem = createTextItemPrefilledWithFileContent(plugin, pluginField);
+            formItemsReplacingNonTextualFormItems.add(previousValueItem);
+
+            chooseCredentialsFormItem = createUploadItemDisabled(plugin, pluginField);
+
+        } else {
+
+            chooseCredentialsFormItem = new UploadItem(plugin.getPluginName() + pluginField.getName(),
+                                                       pluginField.getName());
+        }
+
+        nodeSourceWindow.addCredentialsPickerIcon(pluginField,
+                                                  formItemsReplacingNonTextualFormItems,
+                                                  chooseCredentialsFormItem);
+
+        return formItemsReplacingNonTextualFormItems;
+    }
+
+    private FormItem createUploadItemDisabled(PluginDescriptor plugin, PluginDescriptor.Field pluginField) {
+
+        FormItem chooseCredentialsFormItem;
+        chooseCredentialsFormItem = new UploadItem(plugin.getPluginName() + pluginField.getName(), "");
+        chooseCredentialsFormItem.disable();
+        return chooseCredentialsFormItem;
+    }
+
+    private TextAreaItem createTextItemPrefilledWithFileContent(PluginDescriptor plugin,
+            PluginDescriptor.Field pluginField) {
+
+        TextAreaItem previousValueItem = new TextAreaItem(plugin.getPluginName() + pluginField.getName() +
+                                                          EDIT_FORM_ITEM_SUFFIX, "");
+
+        previousValueItem.setDefaultValue(pluginField.getValue());
+
+        return previousValueItem;
+    }
+
+    private RadioGroupItem createRadioItemToModifyPluginField(PluginDescriptor plugin,
+            PluginDescriptor.Field pluginField) {
+
+        String formItemPrefixName = plugin.getPluginName() + pluginField.getName();
+
+        LinkedHashMap<String, String> radioOptions = new LinkedHashMap<>();
```

Comment:
I guess there is a reason to use 'LinkedHashMap' instead of 'Map'?

Reply:
Yes, this is the order that is used to display it.